cmd/start.go

```diff
@@ -183,3 +191,94 @@ func initHealthServer(s *grpc.Server) {
 	gRPCHealthServer.SetServingStatus("vatz-health-status", healthpb.HealthCheckResponse_SERVING)
 	healthpb.RegisterHealthServer(s, gRPCHealthServer)
 }
+
+type PrometheusManager struct {
+	Protocol string
+	// Contains many more fields not listed in this example.
+}
+
+type PrometheusManagerCollector struct {
+	PrometheusManager *PrometheusManager
+}
+
+type PrometheusValue struct {
+	Up   int
+	Name string
+}
+
+func initPrometheus(port, protocol string) error {
+	log.Info().Str("module", "main").Msgf("Prometheus port: %s", port)
+
+	reg := prometheus.NewPedanticRegistry()
+
+	NewPrometheusManager(protocol, reg)
+
+	reg.MustRegister(
+		prometheus.NewProcessCollector(prometheus.ProcessCollectorOpts{}),
+	)
+
+	http.Handle("/metrics", promhttp.HandlerFor(reg, promhttp.HandlerOpts{}))
+
+	err := http.ListenAndServe(":"+port, nil)
+
+	if err != nil {
+		log.Error().Str("module", "main").Msgf("Prometheus Error: %s", err)
+	}
+
+	return nil
+}
+
+func NewPrometheusManager(protocol string, reg prometheus.Registerer) *PrometheusManager {
```

Comment:
What happens if you try to create multiple PrometheusManager?

Logically the instance of PrometheusManager should exist only one, so the instance should be created through singleton.

Reply:
Thank you for letting me.
I changed by using singleton.
> Commit: